seed/data_importer/tasks.py

```diff
@@ -1325,20 +1323,58 @@ def geocode_and_match_buildings_task(file_pk):
         .iterator()
     )
 
-    id_chunks = [[obj.id for obj in chunk] for chunk in batch(property_states, 100)]
+    # If multiple cycle upload, split properties by cycle
+    # and map each cycle individually
+    # Todo: add a status to import_file to indicate multiple cycle upload
+    # maybe add new source_type Assessed Raw Multiple Cycle?
+    if import_file.multiple_cycle_upload:
+        # Create a dictionary to store the property_state_ids_by_cycle.
+        property_state_ids_by_cycle = defaultdict(list)
+
+        # Loop through the property_state objects.
+        for property_state in property_states:
+            # Find the cycle that corresponds with property_state year_ending.
+
+            # Find a cycle that start <= year_ending <= end
+            if property_state.year_ending:
+                cycle = Cycle.objects.filter(
+                    end__gte=property_state.year_ending,
+                    start__lte=property_state.year_ending,
+                    organization_id=property_state.organization_id
+                ).first()
+            # Check if cycle is none
+            if cycle is None:
+                cycle = Cycle.objects.get(
+                    pk=import_file.cycle_id
+                )
```

Comment:
Isn't there a good chance this is "year_ending"?

Reply:
No, this will be the default cycle now. I'm no longer using "year_ending" as a flag for cycle_id.